polar-core/src/macros.rs

```diff
@@ -198,6 +199,20 @@ macro_rules! rule {
             params,
             body: term!(op!(And)),
             source_info: $crate::sources::SourceInfo::Test,
+            required: false,
+        }
+    }};
+    // this macro variant is used exclusively to create rule *types*
+    // TODO: @patrickod break into specific-purpose rule_type! macro and RuleType struct
+    ($name:expr, [$($args:tt)*], $required:expr) => {{
+        let mut params = args!($($args)*);
+        params.reverse();
+        Rule {
+            name: sym!($name),
+            params,
+            body: term!(op!(And)),
+            source_info: $crate::sources::SourceInfo::Test,
```

Comment:
This source info doesn't seem correct. This isn't coming from a Test.

Reply:
This is a problem with all the macros. They were originally built for tests, but are a little too convenient and have grown to other uses. I created a task to address this.

Reply:
I understand not wanting to fix the existing issue as part of this PR, but I'm not sure how I feel about exacerbating it (adding a new macro with `SourceInfo::Test` and using it outside of tests)

Reply:
follow up